integration/spark/src/main/common/java/io/openlineage/spark/agent/lifecycle/OpenLineageRunEventBuilder.java

```diff
@@ -452,16 +452,24 @@ private RunEvent populateRun(
 
   private <T> Stream<T> buildDatasets(
       List<Object> nodes, Collection<PartialFunction<Object, List<T>>> builders) {
-    PartialFunction<Object, List<T>> fn = PlanUtils.merge(builders);
     return nodes.stream()
         .flatMap(
-            event -> {
-              if (fn.isDefinedAt(event)) {
-                return fn.apply(event).stream();
-              } else {
-                return Stream.empty();
-              }
-            });
+            event ->
+                builders.stream()
+                    .filter(
+                        builder -> {
+                          try {
+                            log.info(
+                                "{} isDefinedAT {} returned {}",
+                                builder.getClass(),
+                                event.getClass(),
+                                builder.isDefinedAt(event));
```

Comment:
At best, this is a debug statement, but I don't really know that it should be here at all